doc/install.rst

```diff
@@ -1,58 +1,655 @@
-Installing
-##############
+Installing Arbor
+################
 
-Installation guide.
+Installation of Arbor is done by obtaining the source code and compiling it on
+the target system.
+
+This guide starts with an overview of the building process, and the various options
+available to customize the build.
+The guide then covers installation and running on `HPC clusters <cluster_>`_, followed by a
+`troubleshooting guide <troubleshooting_>`_ for common build problems.
 
 .. _install_requirements:
 
-Before starting
-===============
+Requirements
+============
+
+Minimum Requirements
+--------------------
+
+The non distributed (i.e. no MPI) version of Arbor can be compiled on Linux or OS X systems
+with very few tools.
+
+.. table:: Required Tools
+
+    =========== ============================================
+    Tool        Notes
+    =========== ============================================
+    git         To check out the code, min version 2.0.
+    cmake       To set up the build, min version 3.0.
+    compiler    A C++11 compiler. See `compilers <compilers_>`_.
+    =========== ============================================
+
+
+.. _compilers:
+
+Compilers
+~~~~~~~~~
+
+Arbor requires a C++ compiler that fully supports C++11 (we have plans to move
+to C++14 soon).
+We recommend using GCC or Clang, for which Arbor has been tested and optimised.
+
+.. table:: Supported Compilers
+
+    =========== ============ ============================================
+    Compiler    Min version  Notes
+    =========== ============ ============================================
+    GCC         5.2.0        5.1 probably works, 5.0 doesn't.
+    Clang       4.0          Clang 3.8 and later probably work.
+    Apple Clang 9
+    Intel       17.0.1       Needs GCC 5 or later for standard library.
+    =========== ============ ============================================
+
+.. Note::
+    The ``CC`` and ``CXX`` environment variables are used to specify the compiler executable
+    to the CMake build scripts. If these are not set, CMake will automatically choose a compiler.
+    On the system that the test below was performed, if the ``CC`` and ``CXX`` variables are
+    not set, CMake used ``/usr/bin/c++``, which was GCC version 4.8.5.
```

Comment:
The comment is a bit ungrammatical, and without specifying the system, it's oddly vague. Can you say instead that on many systems the default C++ compiler may be too old to compile Arbor, for example, GCC 4.8.5?

Reply:
done.